### PR TITLE
feat: [CDB-1144]: Implemented Nav 2.0 changes for Dashboards

### PR DESCRIPTION
**Pull request by Harness-JKhattak**
### Summary

- Implemented Nav 2.0 changes for Dashboards

NOTE - Divider between "Dashboards" and "Folders" exists in a different branch which will be added after merge

### Screenshots

Dashboard List
Screenshot 2023-08-24 at 13 20 51

Folder Screen
Screenshot 2023-08-24 at 13 20 58

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/ModuleRouteConfig.tsx [hunk 1]
```diff
@@ -119,4 +120,8 @@ export const getAllModeRouteDestination = () => {
   return AllModeRouteDestinations()
 }
 
+export const getDashboardsRouteDestinations = () => {
```
Comment (abhinavrastogi-harness) [inline src/modules/ModuleRouteConfig.tsx:123]: this function is redundant. let's just render `<DashboardRouteDestinations />` where required.

File: src/modules/10-common/RouteDefinitionsV2.ts [hunk 2]
```diff
@@ -74,7 +76,17 @@ const routes = {
...
-  toCEGovernance: withRoute<ModulePathParams>(() => '/governance')
+  toCEGovernance: withRoute<ModulePathParams>(() => '/governance'),
+
+  // dashboards routes
+  toDashboardsBase: withRoute<AccountPathProps>(() => '/dashboards'),
+  toDashboardsFoldersPage: withRoute<AccountPathProps>(() => `/dashboards/folders`),
+  toDashboardsFolder: withAccountId<DashboardFolderPathProps>(
+    ({ folderId }) => `/dashboards/folder/${folderId ? folderId : 'shared'}`
+  ),
+  toDashboardsEmbedPage: withAccountId<DashboardEmbedPathProps>(
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/RouteDefinitionsV2.ts:87]: why are these two routes note using `withRoute`?
if not needed, why is the first 2 routes using `withRoute`?

File: src/modules/10-common/RouteDestinationsV2.tsx [hunk 3]
```diff
@@ -64,6 +66,16 @@ const routesV2 = (): React.ReactElement => {
           </Switch>
         </Layout.Horizontal>
       </Route>
+      <Route path={routes.toMode({ ...accountPathProps, mode: NAV_MODE.DASHBOARDS })}>
+        <Layout.Horizontal>
+          <SideNavV2>
+            <DashboardsSideNav />
+          </SideNavV2>
+          <Container style={{ flex: 1 }}>
+            <Switch>{getDashboardsRouteDestinations()}</Switch>
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/RouteDestinationsV2.tsx:75]: `<Switch><DashboardRouteDestinations /></Switch>`

File: src/modules/45-dashboards/DashboardsRouteDestinations.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      <RouteWithLayout exact path={routes.toDashboardsEmbedPage({ ...dashboardEmbedPathProps })}>
+        <DashboardsPage>
+          <DashboardViewPage />
+        </DashboardsPage>
+      </RouteWithLayout>
+    </>
+  )
+}
+
+export default DashboardRouteDestinations
```
Comment (abhinavrastogi-harness) [inline src/modules/45-dashboards/DashboardsRouteDestinations.tsx:64]: nit: this should just be called `RouteDestinations`. no need for `Dashboard` prefix as it is already scoped in `45-dashboards`.

File: src/modules/45-dashboards/pages/DashboardsCommonHeaderPage.tsx [hunk 5]
```diff
@@ -39,11 +41,9 @@ const DashboardsHeader: React.FC = () => {
     updateTitle(titleArray)
   }, [breadcrumbs, getString, updateTitle])
 
-  return (
-    <Page.Header
-      title={title}
-      breadcrumbs={<NGBreadcrumbs links={breadcrumbs} />}
-      content={
+  const dashboard_navigation: React.ReactNode | undefined = React.useMemo(() => {
```
Comment (abhinavrastogi-harness) [inline src/modules/45-dashboards/pages/DashboardsCommonHeaderPage.tsx:44]:
```suggestion
  const dashboard_navigation: React.ReactNode = React.useMemo(() => {
```

Don't use `undefined` at returntype. Return `null` instead when condition is false.

**Comment by abhinavrastogi-harness** [comment]
Build and standards check are failing because of circular dependencies. Please check.

File: src/modules/10-common/RouteDestinationsV2.tsx (hunk 3 above)
Comment (Harness-JKhattak) [reply src/modules/10-common/RouteDestinationsV2.tsx:75]: When I try to change this I get the following linting error:

`@dashboards/DashboardRouteDestinations' import is restricted from being used by a pattern. Import from higher layer modules is not allowedeslint[no-restricted-imports](https://eslint.org/docs/rules/no-restricted-imports)`

**Comment by Harness-JKhattak** [comment]
> Build and standards check are failing because of circular dependencies. Please check.

@as

**Pull request closed without merging**